protocols/gossipsub/src/types.rs

```diff
@@ -109,11 +111,16 @@ pub(crate) struct PeerConnections {
     pub(crate) topics: BTreeSet<TopicHash>,
     /// The rpc sender to the connection handler(s).
     pub(crate) sender: Sender,
+    /// Don't send messages.
+    pub(crate) dont_send: LinkedHashMap<MessageId, Instant>,
 }
 
 /// Describes the types of peers that can exist in the gossipsub context.
 #[derive(Debug, Clone, PartialEq, Hash, EncodeLabelValue, Eq)]
```

Comment:
[Eagerly implement common traits.](https://rust-lang.github.io/api-guidelines/interoperability.html#c-common-traits) Will also avoid some `clone`s in our own code.
```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Hash, EncodeLabelValue, Eq)]
```